api/clober.ts

```diff
@@ -0,0 +1,108 @@
+import { getAddress } from 'viem'
+
+import { fetchGraphQL } from './utils'
+import { THE_GRAPH_API_KEY } from './constants'
+
+type DepthDto = {
+  price: string
+  amount: string
+  isBid: boolean
+}
+
+type Depth = {
+  price: string
+  amount: string
+  isBid: boolean
+}
+
+type TokenDto = {
+  id: string
+  symbol: string
+  decimals: string
+}
+
+type Token = {
+  address: string
+  symbol: string
+  decimals: string
+}
+
+type MarketDto = {
+  id: string
+  orderToken: string
+  a: string
+  r: string
+  d: string
+  quoteToken: TokenDto
+  baseToken: TokenDto
+  depths: DepthDto[]
+}
+
+type Market = {
+  address: string
+  orderToken: string
+  a: string
+  r: string
+  d: string
+  quoteToken: Token
+  baseToken: Token
+  depths: Depth[]
+}
+
+export type MarketsDto = {
+  markets: MarketDto[]
+}
+
+export async function fetchOrderBook(): Promise<Market[]> {
+  const query = `
+    {
+      markets {
+        id
+        orderToken
+        a
+        r
+        d
+        quoteToken {
+          id
+          symbol
+          decimals
+        }
+        baseToken {
+          id
+          symbol
+          decimals
+        }
+        depths {
+          price
+          amount
+          isBid
+        }
+      }
+    }`
+  const { markets } = await fetchGraphQL<MarketsDto>({
+    endpoint: `https://gateway-arbitrum.network.thegraph.com/api/${THE_GRAPH_API_KEY}/subgraphs/id/FYJ75dFzKgGBC7bbxFHna3E3H7M8FNMH1fdjgofGnooV`,
```

Comment:
Let's find a way not to hardcode the endpoint in case we have to support more networks in the future.

Reply:
There are some awesome npm module (https://github.com/graphprotocol/graph-client)

I just put graphql config in [here](https://github.com/clober-dex/coupon.finance/blob/feat/clober-orderbook/.graphclientrc.yml) and use `await execute(query, {})` with `graphql-tag`